x/campaign/keeper/msg_redeem_vouchers.go

```diff
@@ -57,10 +57,14 @@ func (k msgServer) RedeemVouchers(goCtx context.Context, msg *types.MsgRedeemVou
 			Address:    msg.Account,
 			Shares:     types.EmptyShares(),
 		}
+		err = ctx.EventManager().EmitTypedEvent(&types.EventMainnetAccountCreated{Account: account})
+	} else {
+		err = ctx.EventManager().EmitTypedEvent(&types.EventMainnetAccountUpdated{Account: account})
```

Comment:
I think it would be better to have
```
if found {
...
} else {
...
}
```
At the end of the method so we assign the error right before the return